pkg/schemas/mapper/ingress_backend.go

```diff
@@ -0,0 +1,76 @@
+package mapper
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/values"
+)
+
+type IngressSpec struct{}
+
+func (i IngressSpec) FromInternal(data map[string]interface{}) {
+	if _, ok := data["backend"]; ok && data["defaultBackend"] == nil {
+		data["defaultBackend"] = map[string]interface{}{
+			"targetPort": values.GetValueN(data, "backend", "servicePort"),
+			"serviceId":  values.GetValueN(data, "backend", "serviceName"),
+		}
+		delete(data, "backend")
+	}
+	return
+}
+
+func (i IngressSpec) ToInternal(data map[string]interface{}) error {
+	if backend, ok := data["backend"]; (!ok || backend == nil) && data["defaultBackend"] != nil {
+		data["backend"] = map[string]interface{}{
+			"servicePort": values.GetValueN(data, "defaultBackend", "targetPort"),
+			"serviceName": values.GetValueN(data, "defaultBackend", "serviceId"),
+		}
```

Comment:
should defaultBackend be deleted?

Reply:
No, I'm intentionally duplicating all these fields. This way when the object is serialized to either a v1 Ingress or a v1beta1 Ingress it has the right attributes.

Reply:
I'm open to feedback on a better way to do this, I know this is extremely unpretty.

Reply:
I think we should comment context explaining the reasoning on why we are duplicating fields. This is easier to infer in other places because of conditional wrapper `ServerSupportsIngressV1` but harder to understand in mapper. Also would help in future if someone else tries to update/remove these.

Reply:
done